readthedocs/proxito/middleware.py

```diff
@@ -92,19 +97,21 @@ def add_user_headers(self, request, response):
         The headers added come from ``projects.models.HTTPHeader`` associated
         with the ``Domain`` object.
         """
-        if hasattr(request, 'domain'):
+        unresolved_domain = request.unresolved_domain
```

Comment:
Will the request _always_ have the `unresolved_domain` attribute?

Reply:
Yeah, it will always be defined, but looks like you already figured out :)
https://github.com/readthedocs/readthedocs.org/pull/9982#discussion_r1104831202